### PR TITLE
fix: improve config's default exclude

### PR DESCRIPTION
**Pull request by haikyuu**
This PR fixes https://github.com/vitest-dev/vitest/issues/2172.

- [x] Added to the docs

### CODE REVIEW

File: packages/vitest/src/defaults.ts [hunk 1]
```diff
@@ -1,8 +1,7 @@
 import type { BenchmarkUserOptions, ResolvedCoverageOptions, UserConfig } from './types'
 
 export const defaultInclude = ['**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}']
-export const defaultExclude = ['**/node_modules/**', '**/dist/**', '**/cypress/**', '**/.{idea,git,cache,output,temp}/**']
-
+export const defaultExclude = ['**/node_modules/**', '**/dist/**', '**/cypress/**', '**/.{idea,git,cache,output,temp}/**', 'vite.config.*', 'vitest.config.*']
```
Comment (sheremet-va) [inline packages/vitest/src/defaults.ts:4]:
```suggestion
export const defaultExclude = ['**/node_modules/**', '**/dist/**', '**/cypress/**', '**/.{idea,git,cache,output,temp}/**', '{vite,vitest}.config.*']
```
I think we can actually add more configs, like in `defaultCoverageExcludes`

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vitest/src/defaults.ts (hunk 1 above)
Comment (haikyuu) [reply packages/vitest/src/defaults.ts]: Done, I also added it to the docs since it was missing

File: packages/vitest/src/defaults.ts [hunk 2]
```diff
@@ -23,6 +22,7 @@ const defaultCoverageExcludes = [
   '**/__tests__/**',
   '**/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress}.config.{js,cjs,mjs,ts}',
   '**/.{eslint,mocha,prettier}rc.{js,cjs,yml}',
+  '{vite,vitest}.config.*',
```
Comment (sheremet-va) [inline packages/vitest/src/defaults.ts:25]: No, what I meant is to also put files from `:23` line in `defaultExclude` 😄 

23 line already has vite and vitest configs :)

**Review by sheremet-va — CHANGES_REQUESTED** [review]

**Comment by haikyuu** [comment]
Seems like the macos CI machine needs to upgrade its specs since eve≠the linux one is almost 2x faster. And it's probably why the test times out.

**Review by antfu — APPROVED** [review]

**Pull request merged**